src/Form/UserType.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Form;
+
+use App\Entity\User;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\EmailType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Defines the form used to edit an user.
+ *
+ * @author Romain Monteil <monteil.romain@gmail.com>
+ */
+class UserType extends AbstractType
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        // For the full reference of options defined by each form field type
+        // see https://symfony.com/doc/current/reference/forms/types.html
+
+        // By default, form fields include the 'required' attribute, which enables
+        // the client-side form validation. This means that you can't test the
+        // server-side validation errors from the browser. To temporarily disable
+        // this validation, set the 'required' attribute to 'false':
+        // $builder->add('title', null, ['required' => false, ...]);
+
+        $builder
+            ->add('username', TextType::class, [
+                'attr' => ['readonly' => true],
```

Comment:
setting the `readonly` attribute will not enforce this field cannot be submit and username not changed.
You must use [disabled](http://symfony.com/doc/master/reference/forms/types/form.html#disabled) instead.

Reply:
If I use **disabled**, I got an error `Argument 1 passed to App\Entity\User::setUsername() must be of the type string, null given`.

So, what should I do ?
* leave readonly
* add the nullable type to `setUsername()` in `App\Entity\User`
* allow user to change his username
* remove the username input

Reply:
@ker0x, He means a form option, not an HTML attribute:
```php
$builder
    ->add('username', TextType::class,
        'label' => 'label.username',
        'disabled' => true,
    ])
```